src/fund_migration.rs

```diff
@@ -49,11 +156,50 @@ mod tests {
     #[test]
     fn test_signing() {
         let seed = [0u8; 64];
-        let (node_id, signature) = sign_with_ldk_key(seed, "invoice").unwrap();
-        let pub_key = PublicKey::from_str(&node_id).unwrap();
+        let (private_key, public_key) = derive_ldk_keys(&seed).unwrap();
+        let signature = sign_message(&private_key, "invoice");
         let signature = Signature::from_str(&signature).unwrap();
         let message = "I want to payout my funds to invoice".as_bytes();
         let message = Message::from_hashed_data::<sha256::Hash>(message);
-        signature.verify(&message, &pub_key).unwrap();
+        signature.verify(&message, &public_key).unwrap();
+    }
+
+    #[test]
+    #[rustfmt::skip]
+    fn test_lsp_fee() {
+        let lsp_fee = OpeningFeeParams {
+            min_msat: 2_000_000,
+            proportional: 10_000,
+            valid_until: String::new(),
+            max_idle_time: 0,
+            max_client_to_self_delay: 0,
+            promise: String::new(),
+        };
+
+		assert_eq!(add_lsp_fees(      1_000, &lsp_fee),   2_001_000);
+		assert_eq!(add_lsp_fees(  1_000_000, &lsp_fee),   3_000_000);
+		assert_eq!(add_lsp_fees( 11_000_000, &lsp_fee),  13_000_000);
+		assert_eq!(add_lsp_fees(811_000_000, &lsp_fee), 819_191_000);
```

Comment:
According to my calcualtion, this should be:

10_000 ppm = 1%
811_000_000 * 1.01 = 819_110_000

Reply:
As discussed. My assumption here is wrong, because this calculation assumes a payment where the _sent_ amount is an exactly defined number, but what we need for the migration is a payment where the _received_ amount is achieved accurately.